src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ProviderToggleChecker.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using System;
+using System.Security;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal static class ProviderToggleChecker
+    {
+        public static bool IsProviderDisabled()
```

Comment:
This class seems more complex than just inlining. The only value seems to be the safe read of an environment variable. If anything, add an extension to try to read an environment variable, and keep it clear in the caller that we rely on environment variable rather than adding another indirection.

Reply:
Then that helper can be used elsewhere that we read environment variables as well.